### PR TITLE
[TS] LPS-172417

### PR DESCRIPTION
**Pull request by ChrisKian**
https://issues.liferay.com/browse/LPS-172417

Resent from https://github.com/liferay-core-infra/liferay-portal/pull/2979 with changes made to final failure.  Please feel free to squash commits as needed.

Original message:

> When configured with a reverse proxy and javascript/CSS fasload set to true, some resources are not properly retrieved in the combo servlet due to how the resource name is assembled.
> 
> In this situation, a resource name will look something like this:
> com_liferay_server_admin_web_portlet_ServerAdminPortlet:/proxyPath/o/...
> 
> Because we need to remove the proxy path ~~and context path~~, but the modulePortletId comes first, the replacing logic won't work, and our combo files will be incomplete, leading to unexpected errors in the portal.
> 
> My first commit also resolves a small issue where the proxy path is added twice to the DirectServletRegistry path.
> 
> Please let me know if you have any questions, thanks!

### CODE REVIEW

**Comment by ChrisKian** [comment]
SF error seems incorrect, as I cannot find anywhere in our code which matches the expected or actual results.

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by catarineaguiar** [comment]
Formatting issue:

```
   [java] 1: ./portal-impl/src/com/liferay/portal/servlet/DirectServletRegistryImpl.java expected:<...lank(contextPath) &&[ path.startsWith(contextPath)) {]
     [java] 			path = path.subs...> but was:<...lank(contextPath) &&[
     [java] 			path.startsWith(contextPath)) {
     [java] ]
     [java] 			path = path.subs...>
     [java] 
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-01/test-1-26/test-portal-source-format/4547/source-format-current-branch-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by ChrisKian** [comment]
I'm searching for the string `path.startsWith(contextPath)`, but I cannot find it in `DirectServletRegistryImpl.java`.  Retesting sf now.

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@ChrisKian I think it is related to this [commit](https://github.com/liferay/liferay-portal/commit/f6b86b2c69342e90b3b0f543edee87ab00a0f74c#diff-27ec537c61c3966e8fae74382279e8725f3765f1c7b5582bd9bd175978fdbd79L210)

**Comment by ChrisKian** [comment]
Thanks @catarineaguiar,
I can resolve that and force-push changes here if you'd like.

**Comment by catarineaguiar** [comment]
@ChrisKian I would apreciate. Thanks

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by tinatian** [comment]
@ChrisKian Please add test.

Thanks.
Tina

**Comment by ChrisKian** [comment]
Hey @tinatian,
I'm not entirely sure where to add this test, or how to do so tbh.  I see we have a `ComboServletTest` test class, but I don't believe a unit test will properly cover the scenario.  I'm also not sure how to initiate the `service` method from `ComboServlet`.

Any help is appreciated, thanks!

**Comment by tinatian** [comment]
@ChrisKian Unit test should be fine, it can show what kind of parameter will get what kind of request. You can use MockHttpServletRequest and MockHttpServletResponse like all other tests in ComboServletTest

**Comment by ChrisKian** [comment]
Thanks @tinatian,
I've created what I believe to be a valid test, but I'm having a tough time testing locally.  I receive [the same error as CI is getting](https://test-1-3.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/152412/consoleFull) (search for the second entry of `ComboServletTest`).

I believe this may be caused by [LPS-171555](https://issues.liferay.com/browse/LPS-171555), but I am not sure.  Do you have any ideas on how to resolve this error?  Thanks!

**Comment by tinatian** [comment]
@ChrisKian It is setting something not existed, then your option would be : remove the setting and see if anything wrong, then fix anything wrong with the test.

Tina.

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by ChrisKian** [comment]
Hi @tinatian,
Are there any updates with the review?  Thanks!

File: portal-impl/test/unit/com/liferay/portal/servlet/ComboServletTest.java [hunk 1]
```diff
@@ -313,8 +294,6 @@ public boolean isUndeployedPortlet() {
 
 	protected void setUpPortalServletContext() {
 		_portalServletContext = Mockito.spy(new MockServletContext());
-
-		_portalServletContext.setContextPath("portal");
```
Comment (tinatian) [inline portal-impl/test/unit/com/liferay/portal/servlet/ComboServletTest.java:317]: @ChrisKian Why you remove this and add it later?

**Comment by ChrisKian** [comment]
Hi @tinatian, did my comment answer your question?  Should I revise any of my commits?  Thanks!

**Comment by tinatian** [comment]
@ChrisKian Yes, please revise.

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by tinatian** [comment]
see https://github.com/brianchandotcom/liferay-portal/pull/129562

**Pull request closed without merging**